Assets/Scripts/FloatingOrigin/Origin.cs

```diff
@@ -53,25 +53,25 @@ private IEnumerator AttemptShift()
             {
                 yield return new WaitForEndOfFrame();
 
-                var mainShifterPosition = mainShifter.position;
-                var originPosition = transform.position;
-                var newPosition = new Vector3(mainShifterPosition.x, originPosition.y, mainShifterPosition.z);
-                
-                // We use newPosition instead of mainShifterPosition so that the Y component is the same and we 
-                // calculate the distance on a flat plane. This prevents origin shifts when the only change of the
-                // shifter is the height.
-                var distanceBetweenMainShifterAndOrigin = newPosition - originPosition;
-
-                // sqrMagnitude is used because it outperforms magnitude quite a bit: https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html
-                if (distanceBetweenMainShifterAndOrigin.sqrMagnitude < sqrDistanceBeforeShifting)
+                if (((mainShifter.position.x-transform.position.x)* (mainShifter.position.x - transform.position.x))+ ((mainShifter.position.z - transform.position.z) * (mainShifter.position.z - transform.position.z))< sqrDistanceBeforeShifting)
```

Comment:
Graag een methode maken van een kant in een dergelijk complexe condition; ik weet nu wat er staat maar dat gaat vergeten worden. Als je van `((mainShifter.position.x-transform.position.x)* (mainShifter.position.x - transform.position.x))+ ((mainShifter.position.z - transform.position.z) * (mainShifter.position.z - transform.position.z))` een methode maakt met een beschrijvende naam, dan is duidelijk dat het hier gaat om het berekenen van de afstand tussen 2 punten op de XZ plane.

Reply:
methode private int CalculateDistanceSquaredInXZPlane(Vector3 pos1, Vector3 pos2) toegevoegd